script/prerender/contentPages.tsx

```diff
@@ -118,47 +119,63 @@ export const minuteCounter = () => {
 };
 
 let numPages = 0;
-const globalMinuteCounter = minuteCounter();
+export const globalMinuteCounter = minuteCounter();
 export const getStats = () => {
   const elapsedMinutes = globalMinuteCounter();
   return {numPages, elapsedMinutes};
 };
 
-type MakeRenderPage = (services: AppOptions['services']) =>
-  ({code, route}: {route: Match<typeof content>, code: number}) => Promise<SitemapItemOptions>;
-
-const makeRenderPage: MakeRenderPage = (services) => async({code, route}) => {
-
-  if (!route.state || !('bookUid' in route.state)) {
-    throw new Error('match state wasn\'t defined, it should have been');
-  }
+export async function renderAndSavePage(
+  services: AppOptions['services'],
+  savePage: (uri: string, content: string) => void,
+  code: number,
+  serializedMatch: SerializedPageMatch
+) {
+  const match = deserializePageMatch(serializedMatch);
+  const {app, styles, state, url} = await prepareApp(services, match, code);
+  console.info(`Rendering ${url}`); // tslint:disable-line:no-console
 
-  const {bookUid, bookVersion, pageUid} = route.state;
-  const archivePage = assertDefined(
-    await services.archiveLoader.book(bookUid, bookVersion).page(pageUid).load(),
-    'page wasn\'t cached, it should have been'
-  );
+  const html = await renderHtml(styles, app, state);
 
-  const {app, styles, state, url} = await prepareApp(services, route, code);
-  console.info(`rendering ${url}`); // tslint:disable-line:no-console
+  await savePage(url, html);
```

Comment:
do you want to type this on line 130 to return a promise?

Reply:
Ah good idea